lib/sets.bzl

```diff
@@ -40,6 +40,16 @@ def _precondition_only_sets_or_lists(*args):
             fail("Expected arguments to be depset or list, but found type %s: %r" %
                  (t, a))
 
+def _convert_to_list(a):
+    """If the given value is a depset, convert it to a list. Otherwise do nothing.
+
+    Args:
+      *args: The value to be converted
+    """
+    if type(a) == "depset":
+        a = a.to_list()
+    return a
```

Comment:
This file has a broken design (it shouldn't have used `depset` in the first place).
Your fix looks reasonable. An alternative is to keep the code as is. It will break in the future, but this will force users to update to `new_sets` instead.

Reply:
IMO, I'd also prefer we leave it broken to force migration, since it's effectively deprecated now.

Reply:
Is there a plan to rename `new_sets` to `sets`? If so, is that simply going to happen after `sets` is removed?

Ideally we'd break this module with an explicit `fail()` that is sensitive to the Bazel version and/or the breaking `--incompatible_*` flags. Then we can produce a clean error message for the user, instead of frustrating them and leaving it to them to figure out which dependency broke and why. I don't think we expose the necessary info to Starlark at the moment though.

Reply:
I'd prefer that the file be compatible with `--all_incompatible_changes`. If we want people to migrate, then I like removing this file or making it fail based on Bazel version. The `--incompatible_*` warning messages are not terribly helpful for knowing that I need to migrate to `new_sets`. An explicit error message would make it much clearer.

Reply:
Having an explicit error message sounds good.
I don't think it makes sense to maintain this file.